src/main/java/com/yahoo/bullet/querying/Querier.java

```diff
@@ -706,4 +715,18 @@ private void incrementRate() {
     private String getMetaKey() {
         return metaKeys.getOrDefault(Meta.Concept.QUERY_METADATA.getName(), null);
     }
+
+    private void addToTransientFields(PostStrategy postStrategy) {
+        Projection projection = runningQuery.getQuery().getProjection();
+        Aggregation aggregation = runningQuery.getQuery().getAggregation();
+        if (aggregation.getType() == Aggregation.Type.RAW || projection != null) {
+            Map<String, String> projectionFields = projection.getFields();
```

Comment:
NPE if projection is null

Reply:
It's already checked at line 722

Reply:
It was an or before 👍. I guess it is true that we want to do this only if it's RAW and not ALL projection?